### PR TITLE
[feat] added toast notification when user should not be able to create new application

### LINKED ISSUE #178: banner for blocking new apps
yay nice job last week!! 🥦

This week, you’ll be creating a banner that displays when the user clicks on the create application button but is restricted from creating new apps. It will notify them of why they aren’t able to create a new application.

This will essentially function as the user-facing side/frontend for [this sprint by Jinkang](https://github.com/calblueprint/adopt-an-inmate/pull/158) (here’s [the issue](https://github.com/calblueprint/adopt-an-inmate/issues/153) too), so I suggest you review it carefully. You will also be adding a new check to Jinkang’s existing list. Following his example for checks to the `adopter_profiles` table, confirm that the adopter is over 18. 

The basic styling can be [found here (desktop)](https://www.figma.com/design/fPzdPfT77bx4wXHBkAQbj6/Adopt-an-Inmate?node-id=7042-12843) within “Desktop - Dashboard" under where it says “Create New Application Errors” in the green rectangle. Make sure you follow the motion design Jesslyn commented! You can also add two lines of text if some of the messages are long. Here’s all the cases you’ll need to cover (wording might change slightly):
- Please complete your incomplete application before creating a new one.
- You already have two active connections.
- You still have one pending application, please wait for admin approval.
- Your most recent connection is still new, please wait before starting a new application.
- There was an issue with your onboarding responses, please contact admin at adopt@adoptaninmate.org to fix this.
- Adopters must be at least 18 years old before creating an application.
- You have a past rejected application, please follow up with admin at adopt@adoptaninmate.org.

Also restyle the new application button to match the updated designs! Make sure it is faded out but not disabled when adopters are restricted from creating new applications.

*Edit 4/14: added rejected application option and email

### PR DESCRIPTION
**Pull request by saitunguturu**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
-added a toast notification when the user tries to create an application when they should not
-toast fade slides in from right to left, and fades out in place
-greyed out create new button when user satisfied any of these conditions

image

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
It can be reviewed by going through the cases in the screenshot of all the error messages above. The button should only be greyed out when one of these errors is present, and normal when the user should be able to create a new app.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
Main Dashboard should be restyled, right now it is only the Create New button

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @InayaY3

### CODE REVIEW

File: components/MainDashboard/NewApplicationButton.tsx [hunk 1]
```diff
@@ -1,38 +1,85 @@
...
+            height="24"
+            viewBox="0 0 24 24"
+            fill="white"
+          >
+            <path
+              fillRule="evenodd"
+              clipRule="evenodd"
+              d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 5a1 1 0 1 0-2 0v4H7a1 1 0 1 0 0 2h4v4a1 1 0 1 0 2 0v-4h4a1 1 0 1 0 0-2h-4V7z"
+            />
+          </svg>
```
Comment (InayaY3) [inline components/MainDashboard/NewApplicationButton.tsx:80]: mentioned in a prev PR as well, would prefer icons for consistency but not a big deal

File: components/MainDashboard/NewApplicationButton.tsx [hunk 2]
```diff
@@ -1,38 +1,85 @@
...
   return (
-    <Button onClick={createApp} variant="tertiary" disabled={isCreating}>
+    <button
+      onClick={createApp}
+      disabled={isCreating}
+      className="flex cursor-pointer items-center gap-2 rounded-[0.5rem] px-[2.0625rem] py-[0.4375rem] text-white transition-opacity hover:opacity-90"
+      style={{
+        background: isRestricted
+          ? 'linear-gradient(0deg, rgba(255, 255, 255, 0.30) 0%, rgba(255, 255, 255, 0.30) 100%), #AF2028'
+          : '#AF2028',
```
Comment (InayaY3) [inline components/MainDashboard/NewApplicationButton.tsx:57]: aren't some of these colors already in our global styles? i guess we may need to update still.

File: components/MainDashboard/MainDashboard.tsx [hunk 3]
```diff
@@ -8,18 +9,75 @@ export default function MainDashboard() {
...
+            <path
+              d="M5 5L11 11M11 5L5 11"
+              stroke="#E5484D"
+              strokeWidth="1.5"
+              strokeLinecap="round"
+            />
+          </svg>
+          <p className="text-sm text-gray-12">{errorMessage}</p>
+        </div>
+      )}
```
Comment (InayaY3) [inline components/MainDashboard/MainDashboard.tsx:80]: similar to below, would prefer icons? and would also be good to have this as its own component just in case, but not a big deal rn. if this were a bigger sprint/earlier in the sem, would normally fix for you or request changes

**Review by InayaY3 — APPROVED** [review]
lgtm! nice work matching the styles + motion design. left some nit comments regarding styling but should be okay to merge

also updated wording for a few options, and fixed a minor status issue that was messing with some of the checks. make sure you test thoroughly to catch those bugs next time!
